packages/protocol/contracts/governance/DowntimeSlasher.sol

```diff
@@ -136,11 +253,19 @@ contract DowntimeSlasher is SlasherUtil {
     address[] memory groupElectionGreaters,
     uint256[] memory groupElectionIndices
   ) public {
-    address validator = getAccounts().signerToAccount(
-      validatorSignerAddressFromSet(startSignerIndex, startBlock)
+    uint256 startBlock = startBlocks[0];
+    uint256 endBlock = endBlocks[endBlocks.length.sub(1)];
+    require(
+      endBlock.sub(startBlock).add(1) >= slashableDowntime,
+      "the provided intervals must span slashableDowntime blocks"
+    );
+    address validator = getValidatorAccountFromSignerIndex(signerIndices[0], startBlock);
+    require(
+      startBlock > lastSlashedBlock[validator],
+      "cannot slash validator for downtime for which they may already have been slashed"
     );
-    checkIfAlreadySlashed(validator, startBlock);
-    require(isDown(startBlock, startSignerIndex, endSignerIndex), "Not down");
+    require(wasDownForIntervals(startBlocks, endBlocks, signerIndices), "not down");
```

Comment:
I suggest moving this line to the first line of this function and optionally dropping the first two `requires` checks, as they are redundant with the ones in this function. (I could also see an argument for keeping the requires checks for readability)

Reply:
I think this comment was done in the middle of a commit or something.
Correct me if I'm wrong @nategraf but probably this was when I was checking the length of the arrays (startBlocks and endBlocks)